pagopa/io-app
# fix: [IOAPPFD0-63] Add husky git-hook manager with pre-commit and pre-push hooks

## Short description
This PR adds [husky](https://typicode.github.io/husky/#/) as git-hook manager with two hooks:
- `pre-commit`
- `pre-push`

## List of changes proposed in this pull request
- .husky/pre-commit: pre-commit hook using [lint-staged](https://www.npmjs.com/package/lint-staged).
- .husky/pre-commit: custom script to validate that the project lint, prettify and compile status is clean.

## How to test
- Run `yarn prepare` to enable hooks. 
- Then try to commit files that have lint/prettify non-conformities. The commit should not be completed. Prerequisite: remove `--allow-empty` option in `pre-commit` hook to test this or add a `console.log` as an unrecoverable error.
- Commit files with non-conformities with `git commit --no-verify`.
- Try to push to origin. The push should not be completed.
